interfaces/kalosm-common/src/cache.rs

```diff
@@ -194,24 +210,25 @@ impl Cache {
                 let path = self.location.join(model_id).join(revision);
                 let complete_download = path.join(file);
 
-                let api = hf_hub::api::sync::Api::new()?;
                 let repo = Repo::with_revision(
                     model_id.to_string(),
                     RepoType::Model,
                     revision.to_string(),
                 );
-                let api = api.repo(repo);
+                let api = hf_hub::api::sync::Api::new()?.repo(repo);
                 let url = api.url(file);
-                let url = Url::from_str(&url)?;
                 let client = reqwest::Client::new();
+                tracing::trace!("Fetching metadata for {file} from {url}");
                 let response = client
-                    .head(url.clone())
+                    .head(&url)
                     .with_authorization_header(token.clone())
                     .send()
                     .await;
 
                 if complete_download.exists() {
-                    let metadata = tokio::fs::metadata(&complete_download).await?;
+                    let metadata = tokio::fs::metadata(&complete_download).await.map_err(|e| {
+                        CacheError::UnableToGetFileMetadata(complete_download.clone(), e)
+                    })?;
                     let file_last_modified = metadata.modified()?;
```

Comment:
This only fails when the modified field is not available (platform specific error, not a file / permissions error), so adding details about the file doesn't really matter much.